Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3968,6 +3968,10 @@ - (void)resetMappings
             [self.messageMappings updateWithTransaction:transaction];
         }];
         [self updateMessageMappingRangeOptions:MessagesRangeSizeMode_Normal];
+
+        // ViewItem cache may be stale
+        // TODO have a more fine-grained cache expiration based on rows modified.
+        [self.viewItemMap removeAllObjects];
```

Comment:
I suggest we not do this in `resetMappings` but in `setShouldObserveDBModifications`, because `resetMappings` is also used in `loadMoreMessages` and perf will really suffer in the auto-loading experience if we do it here.